t/41-os-autoinst-generate-needle-preview.t

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env perl
+# Copyright 2025 Roland Clobus <rclobus@rclobus.nl>
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+use Test::Warnings ':report_warnings';
+use Mojo::Base -strict, -signatures;
+use File::Compare;
+
+use FindBin '$Bin';
+use lib "$FindBin::Bin/lib", "$Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+
+sub check_os_autoinst_generate_needle_preview_default_needle ($commandline) {
+    system($commandline);
+    is $?, 0, 'Command executed without error code';
+    is compare("$Bin/data/os-autoinst-generate-needle-preview.svg", "$Bin/data/os-autoinst-generate-needle-preview.ref.svg"), 0, 'Equal to reference';
+    unlink("$Bin/data/os-autoinst-generate-needle-preview.svg");
```

Comment:
We generally avoid creating files in the local git checkout. If the test fails and aborts early for some reason, the file is leftover, and our git directories fill up with stuff we don't know where it is coming from.

If you look at other tests in os-autoinst, most of them have this:
```
my $dir = tempdir("/tmp/$FindBin::Script-XXXX");
chdir $dir;
```
And then everything is happening in that directory.

I know that others in my team care less about it, so I'm just saying.